crates/client/rpc-core/src/lib.rs

```diff
@@ -27,6 +27,31 @@ use starknet_core::types::{
 #[derive(Serialize, Deserialize)]
 pub struct Felt(#[serde_as(as = "UfeHex")] pub FieldElement);
 
+/// Starknet write rpc interface.
+#[rpc(server, namespace = "starknet")]
+pub trait StarknetWriteRpcApi {
+    /// Submit a new transaction to be added to the chain
+    #[method(name = "addInvokeTransaction")]
+    async fn add_invoke_transaction(
+        &self,
+        invoke_transaction: BroadcastedInvokeTransaction,
+    ) -> RpcResult<InvokeTransactionResult>;
+
+    /// Submit a new class declaration transaction
+    #[method(name = "addDeployAccountTransaction")]
+    async fn add_deploy_account_transaction(
+        &self,
+        deploy_account_transaction: BroadcastedDeployAccountTransaction,
+    ) -> RpcResult<DeployAccountTransactionResult>;
+
+    /// Submit a new deploy account transaction
+    #[method(name = "addDeclareTransaction")]
+    async fn add_declare_transaction(
+        &self,
+        declare_transaction: BroadcastedDeclareTransaction,
+    ) -> RpcResult<DeclareTransactionResult>;
+}
+
 /// Starknet rpc interface.
 #[rpc(server, namespace = "starknet")]
 pub trait StarknetRpcApi {
```

Comment:
I know the Starknet spec has it like this but will it make it more readable if we call this `StarknetReadRpcApi`?

Reply:
Good idea. I will do the change